R/slice.R

```diff
@@ -42,6 +48,34 @@
 #' x
 #'
 #' vec_slice(mtcars, 1:3)
+#'
+#'
+#' # Type stability --------------------------------------------------
+#'
+#' # The assign variant is type stable. It always returns the same
+#' # type as the input.
+#' x <- 1:5
+#' vec_slice(x, 2) <- 20.0
```

Comment:
What happens if you do `vec_slice(x, 2) <- 20.5`? You get a warning from the cast?

Reply:
This is a lossy cast error that you can lift with `allow_lossy_cast()`.

Reply:
This will surely trip people in general. Maybe we can introduce a custom lossy cast error to suggest creating sequences with `dbl(1:5)` to allow fractional doubles?

Reply:
Maybe we should have used `real` instead of `dbl`...

Reply:
That behaviour seems fine to me